### ISSUE TITLE
[autoscaler][gcp] wrong values for scheduling in example gcp cluster yaml files

### ISSUE THREAD

**Issue by guoqiao** [ray-project/ray#46248]
### What happened + What you expected to happen

In `python/ray/autoscaler/gcp/*.yaml` files, the `scheduling` field is configured as following in a few places:
```
scheduling:
  - preemptible: true
  - onHostMaintenance: TERMINATE
```
The equivalent json version will be:
```
"scheduling": [
    {"preemptilbe": true},
    {"onHostMaintenance": "TERMINATE"}
]
```
This is wrong, according to [GCP Doc](https://cloud.google.com/compute/docs/reference/rest/v1/instances):
```
"scheduling": {
    "onHostMaintenance": enum,
    "automaticRestart": boolean,
    "preemptible": boolean,
    ...
}
```

Summary: for gcp, `scheduling` should be object/dict, but misconfigured to be a list of objects.

### Versions / Dependencies

all versions up to lastest (2.24)

### Reproduction script

in repo root:
```
cd python/ray/autoscaler/gcp/
grep -A 3 "scheduling:" *.yaml
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by anyscalesam** [comment]
Good catch @guoqiao > can you submit a PR and we'll shepherd it through

**Comment by guoqiao** [comment]
@anyscalesam   PR created ^. Thank you!
